### ISSUE TITLE
Field collapsing strange behavior with collapse on a field on a nested object

### ISSUE THREAD

**Issue by jtreher** [elastic/elasticsearch#47117]
**Elasticsearch version** 6.4

**Plugins installed**: Some custom plugins for synonym expansion and hierarchy searches

**JVM version**: 1.8

**OS version** Linux

**Description of the problem including expected versus actual behavior**:

We are attempting to collapse on a field in a nested object. The result is very strange where only a single hit is returned.

If field collapse cannot work on a nested field, I believe that an error should be thrown to avoid confusion.

**Steps to reproduce**:

Example query. Assume the categories field is nested and contains a numeric id field:

POST _search
{
  "query": {
    "match": {"description" : "black"}
  },
  "collapse": {
    "field": "categories.id"
  },
  "_source": "categories.id"
}

Example response:

{
  "took": 6,
  "timed_out": false,
  "_shards": {
    "total": 3,
    "successful": 3,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 27235,
    "max_score": 2.5688996,
    "hits": [
      {
        "_index": "products",
        "_type": "products",
        "_id": "128",
        "_score": 2.5688996,
        "_source": {
          "categories": [
            {
              "id": 419
            },
            {
              "id": 427
            },
            {
              "id": 428
            },
            {
              "id": 567
            }
          ]
        }
      }
    ]
  }
}

**Comment by jimczi** [comment]
I agree that an error should be thrown if possible. Also note that field collapsing doesn't work if the field is multi-valued so this example wouldn't work even if the field is not part of a nested object.

**Comment by joshhornby** [comment]
Does collapsing work on nested objects at all? I'm trying to collapse on a `keyword` type and I'm only seeing one result returned.

**Comment by nkovacic** [comment]
Having the same issue, it would be better if error is thrown that it does not work on nested and multi value fields.

**Comment by marcoserrato** [comment]
Any progress on this? Running on version 7.17, this still seems to be an issue, is this unsupported or is this a bug? Many of my top level documents have a nested document which might share a similar value for a field I'm interested in, I'd like to collapse these results based on that field value in my nested document.
